HW04/Alexey_Zalyotov/spec/my_module_spec.rb

```diff
@@ -0,0 +1,180 @@
+require 'my_module'
+
+class Array
+  include MyModule
+end
+
+# rubocop:disable Metrics/BlockLength
+RSpec.describe '#my_each' do
+  subject { Array.new([1, 2, 3]).my_each(&block) }
+  context 'when no block is given' do
+    let(:block) {}
+    context 'and reciever is array [1, 2, 3]' do
+      it 'returns an Enumerator' do
+        expect(subject).to be_a(Enumerator)
+      end
+      it '.to_a returns array [1, 2, 3]' do
+        expect(subject.to_a).to match_array [1, 2, 3]
+      end
+    end
+
+    context 'or reciever is an empty array' do
+      subject { Array.new([]).my_each(&block) }
+      it 'returns an Enumerator' do
+        expect(subject).to be_a(Enumerator)
+      end
+      it '.to_a returns an empty array' do
+        expect(subject.to_a).to match_array []
+      end
```

Comment:
I prefer to use this way http://www.betterspecs.org/#describe
```suggestion
      describe '#to_a'
        it 'returns an empty array' do
          expect(subject.to_a).to match_array []
        end
      end
```

Reply:
Done